lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1210,6 +1288,408 @@ void BlockedEncodingAttr::print(mlir::AsmPrinter &printer) const {
   printer << "}>";
 }
 
+// FIXME Can we take the LinearLayout by const&?
```

Comment:
Why cannot?

Reply:
This function's definition is generated from the tablegen. I'm not sure how to ask tablegen to generate the `const&`   version.

Reply:
It's probably fine to make a trivial copy. @Mogball may have some suggestions?

Reply:
In general you don't want to pass Attribute, Value or Op per reference as they are reference underneath. Passing by copy is the right thing to do. I would remove the FIXME

Reply:
but this is a `LinearLayout` (not an `Attribute`, but a vector of vectors)